<launch>

  <arg name="model" default="$(env TURTLEBOT3_MODEL)"/>
  <arg name="map_file" default="$(find temizlik_robotu)/maps/house_map.yaml"/>
  <arg name="open_rviz" default="true"/>
  <arg name="move_forward_only" default="false"/>

  <!-- Robot description -->
  <param name="robot_description"
         command="$(find xacro)/xacro '$(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro'" />

  <node pkg="robot_state_publisher"
        type="robot_state_publisher"
        name="robot_state_publisher">
    <param name="publish_frequency" value="50.0"/>
  </node>

  <!-- Map server -->
  <node pkg="map_server"
        type="map_server"
        name="map_server"
        args="$(arg map_file)" />

  <!-- AMCL -->
  <node pkg="amcl"
        type="amcl"
        name="amcl"
        output="screen">
    <param name="use_map_topic" value="true"/>
    <param name="base_frame_id" value="base_link"/>
    <param name="odom_frame_id" value="odom"/>
    <param name="global_frame_id" value="map"/>
    <param name="scan_topic" value="scan"/>
    <param name="min_particles" value="500"/>
    <param name="max_particles" value="2000"/>
  </node>

  <!-- move_base -->
  <include file="$(find temizlik_robotu)/launch/move_base.launch">
    <arg name="model" value="$(arg model)" />
    <arg name="move_forward_only" value="$(arg move_forward_only)"/>
  </include>

  <!-- Waypoint marker -->
  <node pkg="temizlik_robotu"
        type="waypoint_markers.py"
        name="waypoint_markers"
        output="screen">
    <param name="mission_file"
           value="$(find temizlik_robotu)/config/mission.yaml"/>
  </node>

  <!-- RViz -->
  <group if="$(arg open_rviz)">
    <node pkg="rviz"
          type="rviz"
          name="rviz"
          args="-d $(find temizlik_robotu)/rviz/navigation.rviz"/>
  </group>

</launch>

